samples/bluetooth/direct_test_mode/src/dtm.h

```diff
@@ -15,358 +15,265 @@
+ * The CTE is turned off.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int dtm_setup_reset(void);
 
-	/* Physical channel number must be in the range 0..39. */
-	DTM_ERROR_ILLEGAL_CHANNEL,
+/** @brief Set the PHY for DTM.
+ *
+ * @param[in] phy The phy to be used.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int dtm_setup_set_phy(enum dtm_phy phy);
 
-	/* Sequencing error: Command is not valid now. */
-	DTM_ERROR_INVALID_STATE,
+/** @brief Set the modulation for DTM.
+ *
+ * @param[in] modulation The modulation to be used.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int dtm_setup_set_modulation(enum dtm_modulation modulation);
 
-	/* Payload size must be in the range 0..37. */
-	DTM_ERROR_ILLEGAL_LENGTH,
+/** @brief Read supported BLE features.
+ *
+ * @return Features supported by the device.
+ */
+struct dtm_supp_features dtm_setup_read_features(void);
 
-	/* Parameter out of range (legal range is function dependent). */
-	DTM_ERROR_ILLEGAL_CONFIGURATION,
+/** @brief Read the maximum supported parameter value by DTM.
+ *
+ * @param[in] parameter Value to be read.
+ * @param[out] max_val The pointer to the maximum value.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int dtm_setup_read_max_supported_value(enum dtm_max_supported parameter, uint16_t *max_val);
 
-	/* DTM module has not been initialized by the application. */
-	DTM_ERROR_UNINITIALIZED,
-};
+/** @brief Setup the CTE for DTM.
+ *
+ * @param[in] type The CTE type to be used.
+ * @param[in] time The time of CTE in 8us units.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int dtm_setup_set_cte_mode(enum dtm_cte_type type, uint8_t time);
 
-/* The DTM maximum wait time for the UART command second byte. */
-#define DTM_UART_SECOND_BYTE_MAX_DELAY 5
+/** @brief Set the CTE slots duration for DTM.
+ *
+ * @param[in] slot The CTE slots duration.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int dtm_setup_set_cte_slot(enum dtm_cte_slot_duration slot);
 
-/**@brief Function for initializing DTM module.
+/** @brief Set the antenna parameters for DTM.
  *
- *  @return 0 in case of success or negative value in case of error
+ * @param[in] count The antenna count.
+ * @param[in] pattern The antenna switching pattern.
+ * @param[in] pattern_len The length of the pattern.
+ *
+ * @return 0 in case of success or negative value in case of error.
  */
-int dtm_init(void);
+int dtm_setup_set_antenna_params(uint8_t count, uint8_t *pattern, uint8_t pattern_len);
 
-/**@brief Function for waiting between the UART poll cycles.
+/** @brief Set the transmit power for DTM.
+ *
+ * @param[in] power The transmit power request.
+ * @param[in] val TX power value (in dBM) in case of DTM_TX_POWER_REQUEST_VAL request.
+ * @param[in] channel The channel to adjust power (set to 0 if unknown).
```

Comment:
```suggestion
 * @param[in] power   The transmit power request.
 * @param[in] val     TX power value (in dBM) in case of DTM_TX_POWER_REQUEST_VAL request.
 * @param[in] channel The channel to adjust power (set to 0 if unknown).
```